internal/osutils/lockfile/pidlock.go

```diff
@@ -94,6 +96,28 @@ func (pl *PidLock) Close(keepFile ...bool) error {
 	return nil
 }
 
+// WaitForLock will attempt to acquire the lock for the duration given
+func (pl *PidLock) WaitForLock(timeout time.Duration) error {
+	lockedErr := &AlreadyLockedError{}
+	timer := time.NewTimer(timeout)
+	for {
+		_, err := pl.TryLock()
+		if err != nil {
+			if !errors.As(err, &lockedErr) {
+				return errs.Wrap(err, "Unexpected error attempting to acquire lock file")
+			}
+
+			select {
+			case <-timer.C:
+				return err
+			default:
+				time.Sleep(1 * time.Second)
```

Comment:
A lock should generally clear in milliseconds. This is wasting an entire second waiting to try again. Let's run this every 100ms.